hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/dml/TestPartialUpdateForMergeInto.scala

```diff
@@ -66,6 +64,17 @@ class TestPartialUpdateForMergeInto extends HoodieSparkSqlTestBase {
     testPartialUpdateWithInserts("mor", "parquet")
   }
 
+  test("Test partial update with schema on read enabled") {
```

Comment:
I didn't think it was worth it for CI time to test the other cases:
The first 2 just check a table config and we have a ton of confidence that table config works correctly.
The enum one should have the validation unit tested so I'll add that.
And the one I did test was spark config propagation which we are less confident on.

Reply:
OK, quickly test them locally to make sure there is no surprise.

Reply:
In test bootstrap read I added on line 86 
```
sparkSession.sql("create table my_table using HUDI location '" + bootstrapTargetPath + "'");
    sparkSession.sql("select * from my_table").show(5,false);
    sparkSession.sql("merge into my_table as target using (select '92b7d667-cc77-4462-9c05-fcd809ab3aec' as _row_key, 0.51 as begin_lat, 0 as timestamp) source on source._row_key = target._row_key when matched then update set begin_lat = source.begin_lat");
```
and this triggered the bootstrap error:
<img width="1625" alt="image" src="https://github.com/user-attachments/assets/cc74865c-bf9f-4a33-9680-b46fc70683d9">

Reply:
To test enum schema, I used TestHoodieFileGroupReaderOnHive to generate a table with an enum column. Then I copied the table to my tmp dir. Then I ran the spark commands:
```
sparkSession.sql("create table my_table using HUDI location '/tmp/myTable/'");
sparkSession.sql("select * from my_table").show(5,false);
sparkSession.sql("merge into my_table as target using (select '92b7d667-cc77-4462-9c05-fcd809ab3aec' as _row_key, 0.51 as begin_lat, 0 as timestamp) source on source._row_key = target._row_key when matched then update set begin_lat = source.begin_lat");
```
And this triggered the enum error:
<img width="1665" alt="image" src="https://github.com/user-attachments/assets/bec1a8de-75d7-426e-97db-f093bb28b1bc">
But I had to put in a fix to use table schema resolver to get the schema because fullSchema is just a spark schema converted.

Reply:
To test virtual keys I added 
```
options.put(HoodieTableConfig.POPULATE_META_FIELDS.key(), "false");
```
to basicOptions() in TestBootstrapReadBase

Then I ran
```
    sparkSession.sql("create table my_table using HUDI location '" + hudiBasePath + "'");
    sparkSession.sql("select * from my_table").show(5,false);
    sparkSession.sql("merge into my_table as target using (select '92b7d667-cc77-4462-9c05-fcd809ab3aec' as _row_key, 0.51 as begin_lat, 0 as timestamp) source on source._row_key = target._row_key when matched then update set begin_lat = source.begin_lat");
```
and this triggered the error:
<img width="1603" alt="image" src="https://github.com/user-attachments/assets/07fb92a6-6662-4c42-90a5-3a326ccc4abe">